bad_code.java

```diff
@@ -0,0 +1,54 @@
+public class BadCodeExample {
+    
+    public static void main(String[] args) {
+        ChildClass child = new ChildClass();
+        child.doSomething();
+        child.doSomethingElse();
+    }
+}
+
+class ParentClass {
+    public int someNumber = 10;
+    public String someString = "Hello";
+
+    public void doSomething() {
+        System.out.println("Doing something in ParentClass");
+        someNumber++;
+        someString += " Worlds";
+        System.out.println(someString);
+    }
+
+    public void doSomethingElse() {
```

Comment:
In the `doSomethingElse` method of `ChildClass`, consider using a different method or variable names to avoid confusion with the parent class method, especially since they print similar messages. This will improve code readability and maintainability. [medium]